mapentity/views/generic.py

```diff
@@ -171,62 +172,121 @@ def render_to_response(self, context, **response_kwargs):
             response[app_settings['SENDFILE_HTTP_HEADER']] = os.path.join(settings.MEDIA_URL_SECURE, path)
         return response
 
-
-class MapEntityDocument(ModelViewMixin, DetailView):
-    response_class = OdtTemplateResponse
-    with_html_attributes = True
-
-    @classmethod
-    def get_entity_kind(cls):
-        return mapentity_models.ENTITY_DOCUMENT
-
-    def __init__(self, *args, **kwargs):
-        super(MapEntityDocument, self).__init__(*args, **kwargs)
-        # Try to load template for each lang and object detail
-        model = self.get_model()
-        langs = ['_%s' % lang for lang, langname in app_settings['LANGUAGES']]
-        langs.append('')   # Will also try without lang
-
-        def name_for(app, modelname, lang):
-            return "%s/%s%s%s.odt" % (app, modelname, lang, self.template_name_suffix)
-
-        def smart_get_template():
-            for appname, modelname in [(model._meta.app_label, model._meta.object_name.lower()),
-                                       ("mapentity", "mapentity")]:
-                for lang in langs:
-                    try:
-                        template_name = name_for(appname, modelname, lang)
-                        get_template(template_name)  # Will raise if not exist
-                        return template_name
-                    except TemplateDoesNotExist:
-                        pass
-            return None
-
-        found = smart_get_template()
-        if not found:
-            raise TemplateDoesNotExist(name_for(model._meta.app_label, model._meta.object_name.lower(), ''))
-        self.template_name = found
-
-    @view_permission_required()
-    def dispatch(self, *args, **kwargs):
-        return super(MapEntityDocument, self).dispatch(*args, **kwargs)
-
-    def get_context_data(self, **kwargs):
-        rooturl = self.request.build_absolute_uri('/')
-
-        # Screenshot of object map is required, since present in document
-        self.get_object().prepare_map_image(rooturl)
-
-        context = super(MapEntityDocument, self).get_context_data(**kwargs)
-        context['datetime'] = datetime.now()
-        context['STATIC_URL'] = self.request.build_absolute_uri(settings.STATIC_URL)[:-1]
-        context['MEDIA_URL'] = self.request.build_absolute_uri(settings.MEDIA_URL)[:-1]
-        context['MEDIA_ROOT'] = settings.MEDIA_ROOT + '/'
-        if self.with_html_attributes:
-            context['attributeshtml'] = self.get_object().get_attributes_html(self.request)
-        context['objecticon'] = os.path.join(settings.STATIC_ROOT, self.get_entity().icon_big)
-        context['_'] = _
-        return context
+if app_settings['MAPENTITY_WEASYPRINT']:
+    class MapEntityDocument(ModelViewMixin, PDFTemplateResponseMixin, DetailView):
```

Comment:
Oh, no, don't do that.

If you still want to do that, use at least intermediary classes for each implementation.

Would be better to have a ``PRINT_VIEW_CLASS`` setting, or anything else than this pattern :)